File: js/recursionChallenge.js
exports.factorial = function(num) {
    if (num == 0) {
        return 1
    }
        return num * factorial(num - 1)
};

exports.palindrome = function (word) {
    word = word.toString().replace(/[^a-zA-Z0-9]/g, "").toLowerCase()
    word = word.split("")
    
    if (word.shift() != word.pop()) {
        return false
    } else if (word.length <= 1 ) {
        return true
    }
    return palindrome(word)
}

// exports.bottles = function() {

// };

// exports.romanNum = function() {

// };


console.log(palindrome('racecar'))

File: python/recursion_challenge.py
def factorial(num):
	if num == 0:
		return 1
	return num*factorial(num-1)

def palindrome(word):
	word = (str(word)).lower()
	word = "".join(filter(str.isalnum, word))
	word = list(word)
	if word.pop(0) != word.pop(-1): 
		return False
	elif len(word) <= 1:
		return True
	else:	
		return palindrome(word)


def bottles(num):
	pass

def roman_num(num):
	pass

